src/components/OutOfOfficeForm.vue

```diff
@@ -0,0 +1,315 @@
+<!--
+  - @copyright Copyright (c) 2022 Richard Steinmetz <richard@steinmetz.cloud>
+  -
+  - @author Richard Steinmetz <richard@steinmetz.cloud>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<!-- Wait until sieve script has been fetched before showing form. -->
+	<form v-if="sieveScriptData" class="form" @submit.prevent="submit">
+		<div class="form__multi-row">
+			<fieldset class="form__fieldset">
+				<input
+					id="ooo-disabled"
+					class="radio"
+					type="radio"
+					name="enabled"
+					:checked="!enabled"
+					@change="enabled = false">
+				<label for="ooo-disabled">{{ t('mail', 'Vacation responder off') }}</label>
+			</fieldset>
+
+			<fieldset class="form__fieldset">
+				<input
+					id="ooo-enabled"
+					class="radio"
+					type="radio"
+					name="enabled"
+					:checked="enabled"
+					@change="enabled = true">
+				<label for="ooo-enabled">{{ t('mail', 'Vacation responder on') }}</label>
+			</fieldset>
+		</div>
+
+		<div class="form__multi-row">
+			<fieldset class="form__fieldset">
+				<label for="ooo-first-day">{{ t('mail', 'First day') }}</label>
+				<DatetimePicker
+					id="ooo-first-day"
+					v-model="firstDay"
+					:disabled="!enabled" />
+			</fieldset>
+
+			<fieldset class="form__fieldset">
+				<div class="form__fieldset__label">
+					<input
+						id="ooo-enable-last-day"
+						v-model="enableLastDay"
+						type="checkbox"
+						:disabled="!enabled">
+					<label for="ooo-enable-last-day">
+						{{ t('mail', 'Last day (optional)') }}
+					</label>
+				</div>
+				<DatetimePicker
+					id="ooo-last-day"
+					v-model="lastDay"
+					:disabled="!enabled || !enableLastDay" />
+			</fieldset>
+		</div>
+
+		<fieldset class="form__fieldset">
+			<label for="ooo-subject">{{ t('mail', 'Subject') }}</label>
+			<input
+				id="ooo-subject"
+				v-model="subject"
+				type="text"
+				:disabled="!enabled">
+		</fieldset>
+
+		<fieldset class="form__fieldset">
+			<label for="ooo-message">{{ t('mail', 'Message') }}</label>
+			<TextEditor
```

Comment:
https://github.com/nextcloud/mail/pull/6935 changes TextEditor heavily. I recommend to use <textarea> for now ;)